src/common/printer.cpp

```diff
@@ -9,5 +9,21 @@ void Printer::Print(const string &str) {
 	fprintf(stderr, "%s\n", str.c_str());
 #endif
 }
+void ProgressBar::PrintProgress(int percentage) {
+#ifndef DUCKDB_DISABLE_PRINT
+	int lpad = (int)(percentage / 100.0 * pbwidth);
```

Comment:
This function should not be part of the progress bar, but part of the printer, e.g.

`Printer::PrintProgress(int percentage, int width, char *character);`

Then the progress bar can call that using e.g. `Printer::PrintProgress(percentage, PROGRESS_BAR_WIDTH, PROGRESS_BAR_STRING)`